src/main/kotlin/com/mdrobnak/lalrpop/resolve/LpNonterminalReference.kt

```diff
@@ -12,18 +12,16 @@ class LpNonterminalReference(element: LpNonterminalRef) :
     PsiReferenceBase<LpNonterminalRef>(element, TextRange.allOf(element.text)) {
     override fun resolve(): PsiElement? =
         LpResolveUtil.findNonterminal(element.containingFile, element.text).firstOrNull()
-            ?: PsiTreeUtil.getParentOfType(element, LpNonterminal::class.java)
+            ?: element.parentOfType<LpNonterminal>()
                 ?.let { LpResolveUtil.findNonterminalParameter(it, element.text).firstOrNull() }
 
-    override fun handleElementRename(newElementName: String): PsiElement {
-        val newNode = LpElementFactory(element.project).createIdentifier(newElementName)
-        element.firstChild.replace(newNode)
-        return element
+    override fun handleElementRename(newElementName: String): PsiElement = element.apply {
+        val newNode = LpElementFactory(project).createIdentifier(newElementName)
+        firstChild.replace(newNode)
     }
```

Comment:
In contrast to the previous comments about using `apply`, this usage is OK since it's not on `this`.